### DISCUSSION TITLE
When Sending A Notify, The Secondary Queries Itself Instead Of The Primary

### DISCUSSION THREAD

**Discussion by TheDarkula** [PowerDNS/pdns#14071] [Q&A]
I have an autoprimary/autosecondary pair configured. AXFRs function properly, but automatic provisioning is failing.

On the secondary, I ran `pdnsutil add-autoprimary 63.230.56.225 ns2.kincke.com main`

After running `pdns_control notify kincke.com` on the autoprimary, the following logs are emitted:

Primary:
```
Removed from notification list: 'kincke.com' to 63.230.56.226:53 (was acknowledged)
```

Secondary:
```
Received NOTIFY for kincke.com from 63.230.56.226 for which we are not authoritative, trying supermaster
Error resolving SOA or NS for kincke.com at: 63.230.56.226: Query to '63.230.56.226' for SOA of 'kincke.com' produced no answers
```

I am confused, because I would expect the secondary to query `63.230.56.225`, but it instead queries `63.230.56.226`

Am I missing something simple here?

**Comment by Habbie** [comment]
Please see https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open and update your question, thanks.

**Reply by TheDarkula** [reply]
@Habbie All done :)

**Reply by TheDarkula** [reply]
@Habbie  Are you seeing something?

**Comment by omoerbeek** [comment]
I think the secondary retrieves the SOA from the address it received the notify from. Figuring out why the notify is coming from ns2 might help.

**Reply by TheDarkula** [reply]
@omoerbeek That makes sense.
Is there some setting that controls that?

I ran `pdns_control notify kincke.com` on the primary while also running `tcpdump -n 'dst host 63.230.56.226 and port 53' -vvv` on the secondary.

This is the `tcpdump` output:

```
tcpdump: listening on eth0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
08:54:43.109865 IP (tos 0x0, ttl 64, id 25544, offset 0, flags [none], proto UDP (17), length 56)
    10.0.0.165.53 > 63.230.56.226.42879: [bad udp cksum 0x83a2 -> 0xc86b!] 24408 notify*- q: SOA? kincke.com. 0/0/0 (28)
08:54:43.462878 IP (tos 0x0, ttl 64, id 25719, offset 0, flags [DF], proto UDP (17), length 56)
    10.0.0.165.13043 > 63.230.56.226.53: [bad udp cksum 0x83a2 -> 0x686a!] 55270 SOA? kincke.com. (28)
08:54:43.463197 IP (tos 0x0, ttl 64, id 25720, offset 0, flags [none], proto UDP (17), length 56)
    10.0.0.165.53 > 63.230.56.226.42999: [bad udp cksum 0x83a2 -> 0x7360!] 55270 Refused- q: SOA? kincke.com. 0/0/0 (28)
```
